lib/server.js

```diff
@@ -4,6 +4,10 @@ const Hapi = require('hapi');
 const registrationMan = require('./registerPlugins');
 const logger = require('screwdriver-logger');
 
+process.on('unhandledRejection', (reason, p) => {
```

Comment:
Unhandled Promise rejection is getting Deprecated so we should also start adding catch blocks to code blocks where we do new Promise so that we don't end up getting unhandled promise rejection errors.

Reply:
noted. This is a catch all to start identifying areas we are missing proper error handling.